config/areas/settings.php

```diff
@@ -20,6 +20,7 @@
                                     'text'    => $language->name(),
                                 ];
                             }),
+                            'license' => $kirby->user()->isAdmin() ? $kirby->system()->license() : 'Kirby 3',
```

Comment:
@bastianallgeier Isn't `$kirby->system()->license()` already handling this? Only based on the `access.settings` permission instead of admins. I think we shouldn't duplicate this logic, even worse have two different ones.

Reply:
I think Bastian's implementation is fine. Even a user who has access to the settings view should only be able to see the actual license code if they are admin. But you are right – maybe we should change this in the method itself instead.

In any case the fallback `'Kirby 3'` value should only be printed if a license is active (otherwise non-admins would still see a valid license even though the site is not registered):

```php
switch ($license = $kirby->system()->license()) {
    case true:
        $license = 'Kirby 3';
        break;
    case false:
        $license = null;
        break;
}
```

This would also make the prop value more consistent so we don't need [this change](https://github.com/getkirby/kirby/pull/3327/commits/f556c65e931291049f0f6a3b6b650f708ae91a2d).

Reply:
I would solve this within `System::license()` indeed. It's just weird if in that method we look at one permission just to check for another permission when we want to use the method.

Reply:
Then let's change `System::license()` to use the admin property instead of the access permission. I still think it should return `true` if the user is not admin. In the area, the code I posted above can then be used to convert that `true` value into a human-readable "Kirby 3" string.